open-metadata-implementation/access-services/asset-catalog/asset-catalog-server/src/main/java/org/odpi/openmetadata/accessservices/assetcatalog/listenenrs/AssetCatalogOMRSTopicListener.java

```diff
@@ -0,0 +1,209 @@
+import org.odpi.openmetadata.repositoryservices.connectors.stores.metadatacollectionstore.repositoryconnector.OMRSRepositoryHelper;
+import org.odpi.openmetadata.repositoryservices.connectors.stores.metadatacollectionstore.repositoryconnector.OMRSRepositoryValidator;
+import org.odpi.openmetadata.repositoryservices.events.OMRSEventOriginator;
+import org.odpi.openmetadata.repositoryservices.events.OMRSInstanceEvent;
+import org.odpi.openmetadata.repositoryservices.events.OMRSInstanceEventType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+
+/**
+ * AssetCatalogOMRSTopicListener received details of each OMRS event from the cohorts that the local server
+ * is connected to.  It passes indexing events to the publisher.
+ */
+public class AssetCatalogOMRSTopicListener extends OMRSTopicListenerBase
+{
+    private static final Logger log = LoggerFactory.getLogger( AssetCatalogOMRSTopicListener.class );
+    private static final String ASSET_TYPE                         = "Asset";
+
+    private OMRSRepositoryHelper        repositoryHelper;
+    private OMRSRepositoryValidator     repositoryValidator;
+    private String                      serverName;
+    private List<String>                supportedZones;
+    private List<String>                supportedTypesForSearch;
+    private AssetCatalogSearchPublisher publisher;
+    private AssetCatalogConverter       converter;
+
+    public AssetCatalogOMRSTopicListener(String serviceName,
+                                         AuditLog auditLog,
+                                         OpenMetadataTopicConnector outTopicConnector,
+                                         OMRSRepositoryHelper repositoryHelper,
+                                         OMRSRepositoryValidator repositoryValidator,
+                                         String serverName,
+                                         List<String> supportedZones,
+                                         List<String> supportedTypesForSearch
+                                         )
+            throws OMAGConfigurationErrorException {
+            super(serviceName, auditLog);
+            this.publisher = new AssetCatalogSearchPublisher(outTopicConnector);
+            this.serverName = serverName;
+            this.supportedZones = supportedZones;
+            this.repositoryHelper = repositoryHelper;
+            this.repositoryValidator = repositoryValidator;
+            this.converter = new AssetCatalogConverter(serverName, repositoryHelper);
+            this.supportedTypesForSearch = supportedTypesForSearch;
+    }
+
+    /**
+     * Unpack and deliver an instance event to the InstanceEventProcessor
+     *
+     * @param instanceEvent event to unpack
+     */
+    @Override
+    public void processInstanceEvent(OMRSInstanceEvent instanceEvent)
+    {
+        if (log.isDebugEnabled()) {
```

Comment:
Non-functional (insignificant maybe) optiomization we can do: Since we are using slf4j from what I remember if we use parametrized message for debug the check is performed transparently inside the debug method call. Maybe consider:

             `log.debug("Processing instance event: {} ", instanceEvent);`